controllers/clusterenvironment_controller.go

```diff
@@ -177,6 +186,35 @@ func (r *ClusterEnvironmentReconciler) testConnection(ctx context.Context, cfg *
 	return nil
 }
 
+func (r *ClusterEnvironmentReconciler) reconcileServiceNamespaces(ctx context.Context, ce *primazaiov1alpha1.ClusterEnvironment) error {
```

Comment:
To me as of now it's appropriate to store this logic here, but eventually we should move it in the `controlplane` package where we have all the logic for namespace reconciliation.

```suggestion
//TODO: eventually move this logic in `pkg/primaza/controlplane`
func (r *ClusterEnvironmentReconciler) reconcileServiceNamespaces(ctx context.Context, ce *primazaiov1alpha1.ClusterEnvironment) error {
```